examples/qml_features/rust/src/signals.rs

```diff
@@ -49,6 +51,28 @@ pub mod ffi {
             // Emit a signal to QML stating that we have disconnected
             self.as_mut().emit(Connection::Disconnected);
         }
+
+        #[qinvokable]
+        pub fn toggle_logging(mut self: Pin<&mut Self>) {
+            if let Some(connections) = self.as_mut().connections_mut().take() {
+                for conn in connections {
+                    conn.disconnect();
+                }
+            } else {
+                let connections = [
+                    self.as_mut().on_connected(|url| {
+                        println!("Connected: {}", url);
+                    }),
+                    self.as_mut().on_error(|message| {
+                        println!("Error: {}", message);
```

Comment:
These are just function pointers so don't have the ability to move values into them.

So i guess the first argument should always be the C++ context object like in the `queue(|qobject| ...)` method for threading. Otherwise you can't read or manipulate state...

Reply:
Done